### DISCUSSION TITLE
Change the icon shape for task group rows in the grid ui (or otherwise distinguish them?)

### DISCUSSION THREAD

**Discussion by argibbs** [apache/airflow#24414] [Ideas] [answered]
The new grid UI is growing on me (disclaimer: I like marmite) but there are a couple of niggles that make it slightly painful to use.

The TaskGroup nesting is nice, and has led to me aggressively using task groups in my dags. However, there's a couple of things that bug me with the task group rows. The first is already raised as an issue, as it's clearly an actual bug (#24410).

The second I'm raising in the discussions since I wouldn't presume to be the greatest UI guy, but, the problem is that it's really hard to distinguish the task-squares from the group-squares.

Look at the (redacted, sorry) dag below. Quick: In the middle 1100 group, how many (failed) tasks are there? 
![image](https://user-images.githubusercontent.com/4406514/173356314-3d7775e3-0c45-468e-a2f4-fce4f080ce32.png)

I've been using the grid for a while now, and I _still_ see 4 tasks, and 2 failures, but it's actually only 3 total / 1 failure.

![image](https://user-images.githubusercontent.com/4406514/173357197-ec06c490-3a96-4a00-94ce-c2b6ed9af1ba.png)

A primary reason for this confusion is that the "top" task group doesn't get a line above it, so it kinda looks like it belongs to the group above it.

However, I was wondering if (as well as adding the line above it) whether or not it was worth distinguishing the group rows some other way. I was notionally thinging right-angle triangles, e.g. something like (please excuse the crude hand-edit of the screen grab):
![image](https://user-images.githubusercontent.com/4406514/173356618-c13515de-40da-45fe-99d6-a42e9ba1dda8.png)

In this case, it becomes immediately clear which are groups and which are not.

One alternative approach that has occurred to me is to resurrect the outlined/not-outlined behaviour of the old tree view. In the old tree, it was used to distinguish between scheduled/manual dag runs; perhaps here we could outline the group tasks?

Thanks!

**Comment by argibbs** [comment]
One other thing that would help massively (but I assume it's already been raised elsewhere, even though I can't find it), would be to left-align the grid squares, so there wasn't a massive gulf of white space between the row labels and the grid...

**Reply by potiuk** [reply]
Well. I personally think it's not good idea, especially when you have task details panel, you want grid to be close to the panel so that you can move mouse just a litttle from the details panel to swtich tasks.

The task description should take as much as possible (you will see when you have long task names) and when you have bigger number of runs  (which is rather usual) - the whitespace will disappear actualy.

But (just a comment) if you still want to raise it - adding it as commend in an unrelated discussion is probably about the worst way if you REALLY want this to happen.

**Reply by argibbs** [reply]
> adding it as commend in an unrelated discussion is probably about the worst way if you REALLY want this to happen.

\<snorts tea through his nose\> LOL,

Yeah, I'm not that bothered by it - the lines that trace through are mostly enough, and I hadn't thought of it but as you say, there are HCI benefits to having it right aligned. I refer you to my initial comment:
> I wouldn't presume to be the greatest UI guy

I won't mention left-aligning the grid again. :)

**Comment by potiuk** [comment, marked as answer]
I tihnk separating group squares from task squares is something that might be useful (and it's been raised some time with the "border" that @bbovenzi added). I agree that task "box" could be visually distinct. I'd say (and that was my idea before) is not to change the shape but make it look like there are multiple boxes or a bit "bulky" (maybe a subtle bottom-right shadow for the box would be nice) 

But as usual design-by-committee is bad idea and @bbovenzi has an upper-hand with those "ui look & feel" thingies

**Reply by argibbs** [reply]
Ok, so I guess the point is this I'm not the first person to think of this, nor even raise it. So I'll leave it as "top men are on it". Thanks!

**Reply by bbovenzi** [reply]
Yeah, I can work on an improved icon for the task groups and dynamic tasks.

Also, re the right-aligned. I don't think we'd do exactly that. But one of my next up tasks is to make the details panel width adjustable and probably default more space to it so there is less white space in the grid side of things.
